apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -359,30 +362,39 @@ export const getLocalisedSitemap = async (
 export const getResourcePermalinkTree = async (
   siteId: number,
   resourceId: number,
-) => {
+): Promise<string[]> => {
   const resourcePermalinks = await db
     .withRecursive("Ancestors", (eb) =>
       eb
         // Base case: Get the actual resource
         .selectFrom("Resource")
         .where("Resource.siteId", "=", siteId)
         .where("Resource.id", "=", String(resourceId))
-        .select(defaultResourceSelect)
+        .select(["Resource.id", "Resource.permalink", "Resource.parentId"])
         .unionAll((fb) =>
           fb
             // Recursive case: Get all the ancestors of the resource
             .selectFrom("Resource")
             .where("Resource.siteId", "=", siteId)
             .innerJoin("Ancestors", "Ancestors.parentId", "Resource.id")
-            .select(defaultResourceSelect),
+            .select(["Resource.id", "Resource.permalink", "Resource.parentId"]),
         ),
     )
     .selectFrom("Ancestors")
-    .select("Ancestors.permalink")
+    .select(["Ancestors.id", "Ancestors.permalink", "Ancestors.parentId"])
```

Comment:
nit: I think we don't need to select more than the permalink, since it is discarded anyway in the map below.